src/publishApprovedTemplate.ts

```diff
@@ -0,0 +1,36 @@
+import { S3 } from 'aws-sdk'
+import * as fs from 'fs';
+
+import {Arguments} from 'yargs';
+import {loadStackArgs, createStackMain} from './cfn/index';
+import configureAWS from './configureAWS';
+
+export async function publishApprovedTemplate(argv: Arguments): Promise<number> {
+  await configureAWS('sandbox', 'us-east-1');
+
+  const stackArgs = await loadStackArgs(argv);
+  const s3        = new S3();
+
+  if (stackArgs.ApprovedTemplateLocation === undefined ||
+      stackArgs.ApprovedTemplateLocation.length < 0 ) {
+    throw new Error("Missing key: `ApprovedTemplateLocation`")
+  }
+
+  var cfnTemplate = await fs.readFileSync(stackArgs.Template)
+  var bucket      = stackArgs.ApprovedTemplateLocation.substr(5).slice(0,-1)
+  var key         = stackArgs.Template.replace("./","")
+  var params      = { Body: cfnTemplate, Bucket: bucket, Key: key, }
+
+  s3.headBucket({Bucket: bucket}, (err, data) => {
+    if (err) throw new Error(`Invalid Bucket: ${bucket}`)
+  });
```

Comment:
Because JS is async, this error would not get called before `putObject` is called. It's better to use a promise and then block on that promise with `async/await`

```ts
await s3.headBucket({Bucket: bucket}).promise();
```

As an added bonus, this will just throw for you if the bucket doesn't exist and the code after this line won't be executed

Reply:
awesome! I was just about to ask you about this!